pkg/vdri/trustbloc/selection/staticselection/service.go

```diff
@@ -18,9 +20,19 @@ func NewService() *SelectionService {
 	return &SelectionService{}
 }
 
-// SelectEndpoints select endpoints
+// SelectEndpoints select the first endpoint listed for each stakeholder
 func (ds *SelectionService) SelectEndpoints(endpoints []*endpoint.Endpoint) ([]*endpoint.Endpoint, error) {
-	// TODO add logic to select endpoints
-	// For now we just return all endpoints !!! need to be removed after adding selection logic
-	return endpoints, nil
+	var out []*endpoint.Endpoint
+
+	domains := map[string][]*endpoint.Endpoint{}
+
+	for _, ep := range endpoints {
+		domains[ep.Domain] = append(domains[ep.Domain], ep)
+	}
+
+	for _, list := range domains {
```

Comment:
One endpoint is selected from each domain, which means that if we have five orgs then five endpoints are selected - which is overkill. I think you should have a config that limits the number of selected endpoints to N. This would involve randomly selecting N orgs and then randomly selecting an endpoint from each org.